src/lib/redux-cozy-api.js

```diff
@@ -0,0 +1,484 @@
+  const existingDocs = await cozy.client.data.query(index, {
+    selector: { [property]: value },
+    fields: ['_id']
+  })
+  return existingDocs.length === 0
+}
+
+// action creators
+export const registerSchemas = (schemas) => (dispatch) => {
+  dispatch({ type: REGISTER_SCHEMAS, schemas })
+}
+
+const getOrCreateIndex = (doctype) => async (dispatch, getState) => {
+  let index = getIndex(getState(), doctype)
+  if (!index) {
+    const { fields } = getSchema(getState(), doctype)
+    index = await cozy.client.data.defineIndex(doctype, fields)
+    dispatch({ type: GET_OR_CREATE_INDEX, doctype, index })
+  }
+  return index
+}
+
+export const fetchDocuments = (doctype) => async (dispatch, getState) => {
+  try {
+    dispatch({ type: FETCH_DOCUMENTS, doctype })
+    // WARN: cozy-client-js lacks a cozy.data.findAll method that uses this route
+    const resp = await cozy.client.fetchJSON('GET', `/data/${doctype}/_all_docs?include_docs=true`)
+    // WARN: the JSON response from the stack is not homogenous with other routes (offset? rows? total_rows?)
+    // see https://github.com/cozy/cozy-stack/blob/master/docs/data-system.md#list-all-the-documents
+    // WARN: looks like this route returns something looking like a couchDB design doc, we need to filter it:
+    const rows = resp.rows.filter(row => !row.doc.hasOwnProperty('views'))
+    // we normalize the data (note that we add _type so that cozy.client.data.listReferencedFiles works...)
+    const docs = rows.map(row => Object.assign({}, row.doc, { id: row.id, type: doctype, _type: doctype }))
+    // if the doctype references files, we fetch these files IDs here (this is needed for albums)
+    // in the future, if we define more complex relations with other doctypes, we may have to deal
+    // with the retrieval of these association IDs here
+    const relations = getRelations(getState(), doctype)
+    for (let name in relations) {
+      if (relations[name].type === FILES_DOCTYPE) {
+        for (let doc of docs) {
+          doc[name] = await cozy.client.data.listReferencedFiles(doc)
+        }
+      }
+    }
+    // we forge a correct JSONAPI response:
+    const JSONAPIresp = {
+      data: docs,
+      meta: { count: resp.total_rows },
+      skip: resp.offset,
+      next: false
+    }
+    dispatch({ type: RECEIVE_DATA, doctype, response: JSONAPIresp })
+  } catch (error) {
+    dispatch({ type: RECEIVE_ERROR, doctype, error })
+  }
+}
+
+const getOrFetchDocument = (doctype, id) => async (dispatch, getState) => {
+  let doc = getEntity(getState(), doctype, id)
+  if (!doc) {
+    // TODO: handle fetch status & errors with specific reducers:
+    // { entities: 'io.cozy.photos.albums': { entities: {...}, fetchStatuses: {...}, errors: {...} }}
+    dispatch({ type: FETCH_DOCUMENT, doctype, id })
+    doc = await cozy.client.data.find(doctype, id)
+    // we normalize again...
+    doc = Object.assign({}, doc, { id: doc._id, type: doc._type })
```

Comment:
I see [here](https://github.com/cozy/cozy-photos-v3/pull/127/files#diff-a18a75e1e773f60260b32137162523c0R325) that you add `_type` also to the normalized document. Is there a particular reason not to add it here also ?